lib/killbill_client/models/transaction.rb

```diff
@@ -166,7 +172,7 @@ def with_payment_failure_handling
           end
         end
 
-        created_transaction.refresh(options, Payment)
```

Comment:
Looks good overall but out of curiosity, why not pass the `options` to the `with_payment_failure_handling` method? It would prevent having to copy the `refresh` invocation for all methods.